sql: TestDropIndex failed

sql.TestDropIndex [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15287192?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15287192?buildTab=artifacts#/) on master @ [7d5ffef6ab504e6872d660da2b9cb304b936899c](https://github.com/cockroachdb/cockroach/commits/7d5ffef6ab504e6872d660da2b9cb304b936899c):


```
=== RUN   TestDropIndex
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/765430912bd62afa1f42a127d71d4b8a/logTestDropIndex1359436173
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    drop_test.go:492: condition failed to evaluate within 45s: from jobs_verification.go:175: job 2: expected status running, got succeeded
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/765430912bd62afa1f42a127d71d4b8a/logTestDropIndex1359436173
--- FAIL: TestDropIndex (47.46s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDropIndex.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38797

--- a/pkg/sql/drop_test.go
+++ b/pkg/sql/drop_test.go
@@ -497,6 +497,16 @@ func TestDropIndex(t *testing.T) {
 				tableDesc.GetID(),
 			},
 		}); err != nil {
+			// If the job is not running, check if it already succeeded.
+			if secondErr := jobutils.VerifySystemJob(t, sqlRun, 2, jobspb.TypeSchemaChangeGC, jobs.StatusSucceeded, jobs.Record{
+				Username:    username.NodeUserName(),
+				Description: `GC for CREATE INDEX foo ON t.public.kv (v)`,
+				DescriptorIDs: descpb.IDs{
+					tableDesc.GetID(),
+				},
+			}); secondErr == nil {
+				return nil
+			}
 			return err
 		}
 		return nil

